hal/src/main/native/athena/FRCDriverStation.cpp

```diff
@@ -447,10 +445,12 @@ HAL_Bool HAL_IsNewControlData(void) {
   // 20ms rate occurs once every 2.7 years of DS connected runtime, so not
   // worth the cycles to check.
   thread_local int lastCount{-1};
+  if (!dsThread) return false;
+  auto thr = dsThread->GetThread();
+  if (!thr) return false;
   int currentCount = 0;
   {
-    std::unique_lock<wpi::mutex> lock(newDSDataAvailableMutex);
-    currentCount = newDSDataAvailableCounter;
+    currentCount = thr->newDSDataAvailableCounter;
```

Comment:
seems like this could be simplified a bit (e.g. to just `int currentCount = thr->newDSDataAvailableCounter;` and remove the block)